### PR TITLE
Add more parser error tests to validate top level tokens

### PR DESCRIPTION
**Pull request by abhikuhikar**
This PR adds more parser error tests to validate the behavior for the top level tokens which was introduced with the PR #369.  

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by alancai98 — DISMISSED** [review]

File: lang/src/org/partiql/lang/syntax/SqlParser.kt [hunk 1]
```diff
@@ -2647,23 +2647,35 @@ class SqlParser(private val ion: IonSystem) : Parser {
...
             } else {
                 node.throwTopLevelParserError()
             }
         }
-        node.children.map { validateTopLevelNodes(node = it, level = level + 1, topLevelTokenSeen = topLevelTokenSeen || node.type.isTopLevelType) }
+        node.children.map {
+            validateTopLevelNodes(
+                node = it, level = level + 1,
+                topLevelTokenSeen = topLevelTokenSeen || isTopLevelType,
+                dmlListTokenSeen = node.type == DML_LIST
```
Comment (alancai98) [inline lang/src/org/partiql/lang/syntax/SqlParser.kt:2676]: Does this need to also `||` with `dmlListTokenSeen`? i.e.

```kotlin
dmlListTokenSeen = dmlListTokenSeen = dmlListTokenSeen || node.type == DML_LIST
```

All the parser tests pass either way, but just wondering why `topLevelTokenSeen` `||`'s with the accumulator while `dmlListTokenSeen` doesn't.

**Review by alancai98 — CHANGES_REQUESTED** [review]
Just have a few nits and questions. Otherwise lgtm.

File: lang/src/org/partiql/lang/syntax/SqlParser.kt [hunk 2]
```diff
@@ -2647,23 +2647,35 @@ class SqlParser(private val ion: IonSystem) : Parser {
...
                     node.throwTopLevelParserError()
                 }
             } else {
                 node.throwTopLevelParserError()
             }
         }
-        node.children.map { validateTopLevelNodes(node = it, level = level + 1, topLevelTokenSeen = topLevelTokenSeen || node.type.isTopLevelType) }
+        node.children.map {
+            validateTopLevelNodes(
+                node = it, level = level + 1,
```
[the remark this answers is not in the document]
Comment (abhikuhikar) [reply lang/src/org/partiql/lang/syntax/SqlParser.kt:2674]: Changed.

File: lang/src/org/partiql/lang/syntax/SqlParser.kt [hunk 3]
```diff
@@ -2647,23 +2647,35 @@ class SqlParser(private val ion: IonSystem) : Parser {
      * [level] is the current traversal level in the parse tree.
      * If [topLevelTokenSeen] is true, it means it has been encountered at least once before while traversing the parse tree.
      */
-    private fun validateTopLevelNodes(node: ParseNode, level: Int, topLevelTokenSeen: Boolean) {
-        if (topLevelTokenSeen && node.type.isTopLevelType) {
+    private fun validateTopLevelNodes(node: ParseNode, level: Int, topLevelTokenSeen: Boolean, dmlListTokenSeen: Boolean) {
```
[the remark this answers is not in the document]
Comment (abhikuhikar) [reply lang/src/org/partiql/lang/syntax/SqlParser.kt:2650]: Added.

File: lang/src/org/partiql/lang/syntax/SqlParser.kt (hunk 1 above)
Comment (abhikuhikar) [reply lang/src/org/partiql/lang/syntax/SqlParser.kt:2676]: Ahh. Good catch. Thanks. Changed. Wondering how I missed it. 😄

**Review by alancai98 — APPROVED** [review]

**Pull request merged**